pdns/recursor_cache.cc

```diff
@@ -421,6 +420,16 @@ void MemRecursorCache::replace(time_t now, const DNSName &qname, const QType& qt
     }
   }
 
+  if (auth) {
+    /* only if the new entry is auth, we don't want to keep non-auth entry while we have a auth one */
```

Comment:
This comment is hard to read for me. Too much negatives. It is actually  not clear to me atm why we do not want to skip overwriting in the !auth case.

Reply:
Ah, it should be "we don't want to skip overwriting a non-auth entry while we have an auth one"